homeassistant/components/switch/pilight.py

```diff
@@ -26,15 +25,9 @@
 
 DEPENDENCIES = ['pilight']
 
+# Command scheme is arbitrary JSON with at least one entry
 COMMAND_SCHEMA = vol.Schema({
-    vol.Optional(CONF_PROTOCOL): cv.string,
-    vol.Optional('on'): cv.positive_int,
-    vol.Optional('off'): cv.positive_int,
-    vol.Optional(CONF_UNIT): cv.positive_int,
-    vol.Optional(CONF_UNITCODE): cv.positive_int,
-    vol.Optional(CONF_ID): cv.positive_int,
-    vol.Optional(CONF_STATE): cv.string,
-    vol.Optional(CONF_SYSTEMCODE): cv.positive_int,
+    vol.Required(vol.Any(str)): vol.Any(str, int, float)
```

Comment:
That should be fine, because numbers as a protocol key [do not exist](https://manual.pilight.org/en/configuration-devices). Think of it as a python dictionary with string keys and string/int/float values.